mt-kahypar/datastructures/concurrent_flat_map.h

```diff
@@ -0,0 +1,174 @@
+/*******************************************************************************
```

Comment:
at some point (not now) we might want to unify some of the copy-pasta code for all of the hash maps used